File: caesar.py
#!/usr/bin/env python3
from sys import argv
from cs50 import get_string


def main():
    # Ensure correct usage
    if len(argv) != 2 or int(argv[1]) < 0: #or # ins'tnum:
        print("Usage: python caesar.py k or ./caesar.py k")
        return 1
    k = int(argv[1]) % 26
    # Insert text to encrypt
    plaintext = get_string("plaintext: ")
    print("ciphertext: ", end="")
    for c in plaintext:
        # Encrypt every alphabetical character
        # Encrypt upper case characters
        if ord(c) > 64 and ord(c) < 91:
            if ord(c) + k < 91:
                print(chr(ord(c) + k), end="")
            else:
                print(chr(64 + (ord(c) + k) % 90), end="")
        # Encrypt lower case characters
        elif ord(c) > 96 and ord(c) < 123:
            if ord(c) + k < 123:
                print(chr((ord(c) + k)), end="")
            else:
                print(chr(97 + ((ord(c) + k)) % 123), end="")
        # Do not encrypt non-alphabetical characters
        else:
            print(c, end="")
    print()


if __name__ == "__main__":
    main()

File: caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    // Key validation.
    if (argc == 1)
    {
        printf("Usage: ./caesar key\n");
    }
    
    int key = atoi(argv[argc - 1]);    
    if (argc == 2 && key > 0)
    {
        key %= 26;
        printf("Success\n%i\n", key);
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    
    // Ask for the original message.
    string plaintext = get_string("plaintext:  ");
    
    // Ciphering the message.
    printf("ciphertext: ");
    for (int i = 0, n = strlen(plaintext); i < n; i++)
    {
        // For uppercase
        if (plaintext[i] >= 65 && plaintext[i] <= 90)
        {
            if ((plaintext[i] + key) <= 90)
            {
                printf("%c", plaintext[i] + key);
            }
            else
            {
                int loop = (plaintext[i] + key) % 90;
                printf("%c", 64 + loop);  
            }
        }
        // For lower case
        else if (plaintext[i] >= 97 && plaintext[i] <= 122)
        {
            if ((plaintext[i] + key) >= 122)
            {
                int loop = (plaintext[i] + key) % 122;
                printf("%c", 96 + loop);
            }
            else 
            {
                printf("%c", plaintext[i] + key);
            }
        }         
        else
        {
            printf("%c", plaintext[i]);
        }    
    }
    printf("\n");
}
    
